src/components/CoachRegistration.tsx

```diff
@@ -0,0 +1,142 @@
+    sport: string | null,
+    expertise: string | null,
+    password: string | null
+}
+
+export class CoachRegistration extends React.Component<{},CoachRegistrationState> {
+    
+    constructor(props: any) {
+        super(props)
+        
+        this.state = {
+            firstName: null,
+            lastName: null,
+            email: null,
+            phone: null,
+            dob: null,
+            location: null,
+            postcode: null,
+            address1: null,
+            address2: null,
+            town: null,
+            county: null,
+            qualifications: null,
+            yearsCoaching: null,
+            sport: "Rugby",
+            expertise: null,
+            password: null
+        }
+        this.updateFirstName=this.updateFirstName.bind(this)
+        this.updateLastName=this.updateLastName.bind(this)
+        this.updateEmail=this.updateEmail.bind(this)
+        this.updatePhone=this.updatePhone.bind(this)
+        this.updateDateOfBirth=this.updateDateOfBirth.bind(this)
+        this.updateLocation=this.updateLocation.bind(this)
+        this.updatePostcode=this.updatePostcode.bind(this)
+        this.updateAddress1=this.updateAddress1.bind(this)
+        this.updateAddress2=this.updateAddress2.bind(this)
+        this.updateTown=this.updateTown.bind(this)
+        this.updateCounty=this.updateCounty.bind(this)
+        this.updateQualifications=this.updateQualifications.bind(this)
+        this.updateYearsTraining=this.updateYearsTraining.bind(this)
+        this.updateSport=this.updateSport.bind(this)
+        this.updateAreaOfExpertise=this.updateAreaOfExpertise.bind(this)
+        this.updatePassword=this.updatePassword.bind(this)
+        this.sendResults=this.sendResults.bind(this)
+    }
+
+    updateFirstName(newFName: string) {this.setState({firstName: newFName})}
+    updateLastName(newLName: string) {this.setState({lastName: newLName})}
+    updateEmail(newEmail: string) {this.setState({email: newEmail})}
+    updatePhone(newPhone: string) {this.setState({phone: newPhone})}
+    updateDateOfBirth(newDOB: string) {this.setState({dob: newDOB})}
+    updateLocation(newLoc: Location) {this.setState({location: newLoc})}
+    updateAddress1(newAddr1: string) {this.setState({address1: newAddr1})}
+    updateAddress2(newAddr2: string) {this.setState({address2: newAddr2})}
+    updateTown(newTown: string) {this.setState({town: newTown})}
+    updateCounty(newCounty: string) {this.setState({county: newCounty})}
+    updatePostcode(newPS: string) {this.setState({postcode: newPS})}
+    updateQualifications(newQuals: string) {this.setState({qualifications: newQuals})}
+    updateYearsTraining(newYT: number) {this.setState({yearsCoaching: newYT})}
+    updateSport(newSport: string) {this.setState({sport: newSport})}
+    updateAreaOfExpertise(newAOE: string) {this.setState({expertise: newAOE})}
+    updatePassword(newPass: string) {this.setState({password: newPass})}
```

Comment:
Again, if you make these all fat arrow functions you wont have to bind this in the constructor